server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -664,58 +668,90 @@ private void runReadyClone(
         RepositoryShardId repoShardId,
         Repository repository
     ) {
-        final SnapshotId targetSnapshot = target.getSnapshotId();
-        final String localNodeId = clusterService.localNode().getId();
-        if (currentlyCloning.add(repoShardId)) {
-            repository.cloneShardSnapshot(
-                sourceSnapshot,
-                targetSnapshot,
-                repoShardId,
-                shardStatusBefore.generation(),
-                ActionListener.wrap(
-                    generation -> innerUpdateSnapshotState(
-                        new ShardSnapshotUpdate(target, repoShardId, new ShardSnapshotStatus(localNodeId, ShardState.SUCCESS, generation)),
-                        ActionListener.runBefore(
-                            ActionListener.wrap(
-                                v -> logger.trace(
-                                    "Marked [{}] as successfully cloned from [{}] to [{}]",
-                                    repoShardId,
-                                    sourceSnapshot,
-                                    targetSnapshot
-                                ),
-                                e -> {
-                                    logger.warn("Cluster state update after successful shard clone [{}] failed", repoShardId);
-                                    failAllListenersOnMasterFailOver(e);
-                                }
-                            ),
-                            () -> currentlyCloning.remove(repoShardId)
-                        )
-                    ),
-                    e -> innerUpdateSnapshotState(
-                        new ShardSnapshotUpdate(
-                            target,
+        repository.getRepositoryData(new ActionListener<RepositoryData>() {
+            @Override
+            public void onResponse(RepositoryData repositoryData) {
+                try {
+                    RemoteStoreLockManagerFactory remoteStoreLockManagerFactory = null;
+                    final IndexMetadata indexMetadata = repository.getSnapshotIndexMetaData(
+                        repositoryData,
+                        sourceSnapshot,
+                        repoShardId.index()
+                    );
+                    final boolean isRemoteStoreInteropEnabled = repository.getSnapshotInfo(sourceSnapshot)
+                        .isRemoteStoreIndexShallowCopyEnabled();
+                    if (isRemoteStoreInteropEnabled
+                        && indexMetadata.getSettings().getAsBoolean(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, false)) {
+                        remoteStoreLockManagerFactory = new RemoteStoreLockManagerFactory(() -> repositoriesService);
```

Comment:
lets initialize `remoteStoreLockManagerFactory` in constructor of `SnapshotsService` once.